sdk/python/kfp/containers/_component_builder.py

```diff
@@ -214,7 +235,24 @@ def build_python_component(component_func, target_image, base_image=None, depend
                                   ' and push the image to ' +
                                   target_image)
 
-  component_spec = _func_to_component_spec(component_func, base_image=base_image)
+  component_spec = _func_to_component_spec(
+      component_func, base_image=base_image)
+
+  if is_v2:
+    # Annotate the component to be a V2 one.
+    if getattr(component_spec, 'metadata', None) and getattr(
```

Comment:
JFYI: `if getattr(component_spec, 'metadata', None)` can probably be replaced with just `if component_spec.metadata`. The structure field always exist (although they can be None).

Reply:
Maybe we can try to make this code easier to understand:
```python
if not component_spec.metadata:
    component_spec.metadata = _structures.MetadataSpec()
if not component_spec.metadata.annotations:
    component_spec.metadata.annotations = {}

component_spec.metadata.annotations[V2_COMPONENT_ANNOTATION] = 'true'
```

Reply:
Good point! Done.